src/github.rs

```diff
@@ -182,31 +117,94 @@ let token_res = &data.oauth
                 .json(e.to_string());
         }
         }
-        /*match userdetails(access_token.clone()).await {
-            Ok(val) => {    
-                //dbg!(val);
-                dbg!(val.get("login").clone());
-
-            },
-            Err(e) => {
-                dbg!(e);
-            },
-        };*/
-        //access_token.secret().to;
-
     }else{
-        /*let html = format!(
-            r#"<html>
-            <head><title>OAuth2 Test</title></head>
-            <body>
-                Gitlab user info:
-                <pre></pre>
-                <a href="/">Home</a>
-            </body>
-        </html>"#,);
-        actix_web::HttpResponse::Ok().body(html)*/
         return actix_web::HttpResponse::BadRequest().content_type("application/json")
         .json("No user details found.");
     }
 
-}
\ No newline at end of file
+}
+async fn userdetails(access_token:String) -> Result<serde_json::value::Value,awc::error::HttpError> {
```

Comment:
Rename this to `user_details`. In general prefer using `_` to separate words in function or module names.